tests/test_utils.py

```diff
@@ -298,11 +306,536 @@ def check_multistream_classic(fnames):
+
+                    # save plot
+                    plotname = '%s_%s.png' % (lat, lng)
+                    print(lat)
+                    print(lng)
+                    os.makedirs(os.path.join(plot_dir, 'annual_mean'),
+                                exist_ok=True)
+                    savepath = os.path.join(plot_dir, 'annual_mean', plotname)
+                    plt.savefig(savepath, bbox_inches='tight', dpi=dpi)
+                    print("saved figure to %s" % savepath)
+
+                    plt.clf()
+                    plt.close()
+
+            if 'monthly_mean_diurnal_cycle' in plots_to_make:
+
+                # make monthly mean diurnal cycle plots
+                with plt.rc_context(dict(sns.axes_style("whitegrid"),
+                                    **sns.plotting_context("notebook"))):
+                    f, axarr = plt.subplots(4, 12, figsize=(35, 7),
+                                            sharex=True,
+                                            sharey=True)
+
+                    months = ['January', 'February', 'March', 'April', 'May',
+                              'June', 'July', 'August', 'September',
+                              'October', 'November', 'December']
+
+                    for i, (vic_var, variable_name) in enumerate(vars.items()):
+                        for j in range(12):
+
+                            # calculate monthly mean diurnal cycle
+                            monthly_mean = {}
+                            for (key, df) in data.items():
+                                monthly_mean[key] = pd.DataFrame(df[vic_var].
+                                                                 groupby(
+                                                                 [df.index.
+                                                                  month,
+                                                                  df.index.
+                                                                  hour]).
+                                                                 mean())
+
+                            df = pd.DataFrame({key: d[vic_var] for key, d in
+                                              monthly_mean.items() if vic_var
+                                              in d})
+
+                            for key, series in df.iteritems():
+                                series[j+1].plot(linewidth=compare_data_dict
+                                                 [key]['linewidth'],
+                                                 ax=axarr[i, j],
+                                                 color=compare_data_dict[key]
+                                                 ['color'],
+                                                 linestyle=compare_data_dict
+                                                 [key]['linestyle'],
+                                                 zorder=compare_data_dict[key]
+                                                 ['zorder'])
```

Comment:
I've updated this as you suggested before with a smaller hanging indent, but the line is still too short - the `linewidth=compare_data_dict[key]['linewidth']` line still has to be broken up.